boards/arm/arduino_portenta_h7/arduino_portenta_h7-common.dtsi

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Benjamin Björnsson <benjamin.bjornsson@gmail.com>.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	gpio_keys {
+		compatible = "gpio-keys";
+		wake_up: button {
+			label = "wakeup";
+			gpios = <&gpioc 13 GPIO_ACTIVE_HIGH>;
+			status = "disabled";
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		red_led: led_0 {
+			gpios = <&gpiok 5 GPIO_ACTIVE_LOW>;
+			label = "Red LED";
+		};
+		green_led: led_1 {
+			gpios = <&gpiok 6 GPIO_ACTIVE_LOW>;
+			label = "Green LED";
+		};
+		blue_led: led_2 {
+			gpios = <&gpiok 7 GPIO_ACTIVE_LOW>;
+			label = "Blue LED";
+		};
+	};
+
+	aliases {
+		led0 = &red_led;
+		led1 = &green_led;
+		led2 = &blue_led;
+	};
+};
+
+&rcc {
+	d1cpre = <1>;
+	hpre = <2>;
+	d1ppre = <2>;
+	d2ppre1 = <2>;
+	d2ppre2 = <2>;
+	d3ppre = <2>;
+};
+
+/* UART0 in datasheet */
+&uart4 {
+	pinctrl-0 = <&uart4_tx_pa0 &uart4_rx_pi9>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+};
+
+/* UART1 in datasheet */
+&usart1 {
+	pinctrl-0 = <&usart1_tx_pa9 &usart1_rx_pa10>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+};
+
+/* UART2 in datasheet */
+&usart6 {
+	pinctrl-0 = <&usart6_tx_pg14 &usart6_rx_pg9>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+};
+
+/* UART3 in datasheet */
+&uart8 {
+	pinctrl-0 = <&uart8_tx_pj8 &uart8_rx_pj9>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+};
+
+/* I2C0 in datasheet */
+&i2c3 {
+	pinctrl-0 = <&i2c3_scl_ph7 &i2c3_sda_ph8>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+/* I2C1 in datasheet */
+&i2c1 {
+	pinctrl-0 = <&i2c1_scl_pb6 &i2c1_sda_pb7>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+/* I2C2 in datasheet */
+&i2c4 {
+	pinctrl-0 = <&i2c4_scl_ph11 &i2c4_sda_ph12>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+/* I2C3 in datasheet */
+&i2c3 {
+	pinctrl-0 = <&i2c3_scl_ph7 &i2c3_sda_ph8>;
+	pinctrl-names = "default";
+	clock-frequency = <I2C_BITRATE_FAST>;
+};
+
+/* SPI1 in datasheet */
+&spi2 {
+	pinctrl-0 = <&spi2_nss_pi0 &spi2_sck_pi1
+		     &spi2_miso_pc2 &spi2_mosi_pc3>;
+	pinctrl-names = "default";
+};
+
+&can1 {
+	pinctrl-0 = <&fdcan1_rx_pb8 &fdcan1_tx_ph13>;
+	pinctrl-names = "default";
+};
+
+&rtc {
+	status = "okay";
+};
+
+zephyr_udc0: &usbotg_fs {
```

Comment:
I am not very confident, but it seems to me that only the [stm32h743.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/7fe5ab315a4453322780e5655364fe4de2a1b9b6/dts/arm/st/h7/stm32h743.dtsi#L22) file has a definition for the `usbotg_fs`.

On the Portenta, as I have understood the schematic, only the High Speed interface is available without modifying R44 and R45 on the board:
![image](https://user-images.githubusercontent.com/8845664/166299218-af38c9c2-bc7e-4944-a140-511ad303def4.png)

That's why I have opened this issue: https://github.com/zephyrproject-rtos/zephyr/issues/45094

Reply:
To my understanding, the USB HS is the one connected to the boards USB-C connector. This goes through an external PHY which is not yet supported in Zephyr. The USB FS on the other hand is reachable on the Portenta Breakout board, which is what I used to verify the usb/console sample.

Reply:
Ok, I see.
Is it ok to define it here while only available with an extension board?

Reply:
I have tried the usb/console sample on the MachineControl (which is using USB FS like the Breakout board) but the ACM port does not show up on Linux.

Reply:
Have you looked at the output from the UART while running this sample? I've successfully been running both subsys/usb/cdc_acm and subsys/usb/console on my Mac.

Reply:
Sorry, I have misunderstood the MachineControl schematic, only the USB A socket is USB FS, the micro-usb is USB HS.
I could run the subsys/usb/cdc_acm and subsys/usb/console samples on my side, sorry about that.